src/web/mjs/connectors/XlecX.mjs

```diff
@@ -0,0 +1,52 @@
+import Connector from '../engine/Connector.mjs';
+import Manga from '../engine/Manga.mjs';
+
+export default class XlecX extends Connector {
+
+    constructor() {
+        super();
+        super.id = 'xlecx';
+        super.label = 'XlecX';
+        this.tags = [ 'hentai', 'porn', 'english' ];
+        this.url = 'https://xlecx.org';
+    }
+
+    async _getMangaFromURI(uri) {
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchDOM(request, 'main div#dle-content div.full-in h1');
+        return new Manga(this, uri.pathname, data[0].textContent.trim());
+    }
+
+    async _getMangas() {
+        let mangaList = [];
+        const request = new Request(new URL(this.url), this.requestOptions);
+        const data = await this.fetchDOM(request, 'div.pagi-nav div.navigation a:nth-last-of-type(2)');
+        const pageCount = parseInt(data[0].text.trim());
+        for(let page = 1; page <= pageCount; page++) {
+            const mangas = await this._getMangasFromPage(page);
+            mangaList.push(...mangas);
+        }
+        return mangaList;
+    }
+
+    async _getMangasFromPage(page) {
+        const request = new Request(new URL(`/page/${page}/`, this.url), this.requestOptions);
+        const data = await this.fetchDOM(request, 'main div#dle-content div.thumb a.th-title');
+        return data.map(element => {
+            return {
+                id: this.getRootRelativeOrAbsoluteLink(element, this.url),
+                title: element.text.trim()
+            };
+        });
+    }
+
+    async _getChapters(manga) {
+        return [ Object.assign({ language: '' }, manga) ];
```

Comment:
Same comment as gntai #2751 . how is that _getChapters working ?

Reply:
Hentai websites are usually one-shots without chapters, the manga is basically the one shot chapter.
It creates a new object which acts as chapter containing a `language` property and then assigned the `id` and `title` property from the manga to it.

Reply:
Samples for clarification of `Object.assign`:

```javascript
let source = { b: 99, x: 0 };
let target = { a: 1, b: 2 };
Object.assign(target, source);
console.log(source, target);
```

```javascript
let manga = { id: '/comic/my-little-pony', title: 'My Big Pony' };
let chapter = { language: 'pony' };
Object.assign(chapter, manga);
console.log(chapter);
```